api/client/samples/similar_regions/similar_region.py

```diff
@@ -444,8 +444,9 @@ def map_response(idx, _, response, *args):
         self._logger.warning("Getting data series for {} regions in {} batch(es) for property {}".
                           format(n_reg, n_queries, prop_name))
         load_bar = tqdm(total=n_reg)
-        # Sending out all queries
-        self.client.batch_async_get_data_points(queries, map_result=map_response)
+        # Sending out all queries -- use fresh client instance to avoid mixing async connection pools between runs
+        data_client = GroClient(API_HOST, ACCESS_TOKEN)
```

Comment:
Just to make sure I understand - this data_client is local and separate from self.client. This is only used for this particular data download (i.e. this particular property and region list). I don't think I saw "confusing messages" you are referring to and not sure how having a separate client would help. But if it is I guess this is fine